### PR TITLE
[FSPE-7526] Add support of multiple date selection to date picker web component

### PR DESCRIPTION
**Pull request by janibolkvadze**
https://ctl.atlassian.net/browse/FSPE-7526

@dani-cl-madrid @jllr @mattnickles

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/1/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/2/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://c007bf77ea27a27a8e05bb9c66b9760003cc7ce0-chi-pr.enterprise.ctl.io

File: src/custom-elements/docs/docs.json [hunk 1]
```diff
@@ -699,6 +699,23 @@
           "optional": false,
           "required": false
         },
+        {
+          "name": "multiple",
+          "type": "boolean",
+          "mutable": false,
+          "attr": "multiple",
+          "reflectToAttr": true,
+          "docs": "To allow the user selecting multiple dates",
```
Comment (mattnickles) [inline src/custom-elements/docs/docs.json:708]:
```suggestion
          "docs": "To allow the user to select multiple dates",
```

File: src/custom-elements/docs/docs.json [hunk 2]
```diff
@@ -953,6 +970,23 @@
...
```
Comment (mattnickles) [inline src/custom-elements/docs/docs.json:979]:
```suggestion
          "docs": "To allow the user to select multiple dates",
```

File: src/custom-elements/src/components/date-picker/date-picker.tsx [hunk 3]
```diff
@@ -60,6 +59,11 @@ export class DatePicker {
    */
   @Prop({ reflect: true }) mode: DatePickerModes = 'date';
 
+  /**
+   * To allow the user selecting multiple dates
```
Comment (mattnickles) [inline src/custom-elements/src/components/date-picker/date-picker.tsx:63]:
```suggestion
   * To allow the user to select multiple dates
```

File: src/custom-elements/src/components/date/date.tsx [hunk 4]
```diff
@@ -60,6 +60,11 @@ export class Date {
    */
   @Prop({ reflect: true }) excludedDates: string;
 
+  /**
+   * To allow the user selecting multiple dates
```
Comment (mattnickles) [inline src/custom-elements/src/components/date/date.tsx:64]:
```suggestion
   * To allow the user to select multiple dates
```

File: src/website/views/components/date-picker/_examples.pug [hunk 5]
```diff
@@ -202,6 +202,33 @@ p.-text
       <chi-date-picker id="date-time" mode="datetime"></chi-date-picker>
   #html-datetime.chi-tabs-panel
 
+h3 Multiple dates
+p.-text Use the attribute <code>multiple</code> to allow the user selecting multiple dates
```
Comment (mattnickles) [inline src/website/views/components/date-picker/_examples.pug:206]:
```suggestion
p.-text Use the attribute <code>multiple</code> to allow the user to select multiple dates.
```

**Review by mattnickles — CHANGES_REQUESTED** [review]
Minor copy suggestion

File: src/custom-elements/src/components/date/date.tsx [hunk 6]
```diff
@@ -134,7 +149,11 @@ export class Date {
    */
   @Method()
   async setDate(date) {
-    this.value = date;
+    if (this.multiple) {
+      this.value = `${this.value ? this.value + ',' : ''}${date}`;
+    } else {
+      this.value = date;
+    }
   }
```
Comment (dani-cl-madrid) [inline src/custom-elements/src/components/date/date.tsx:157]: I would keep set as set and maybe create a new method for `addDate`

File: src/custom-elements/src/components/date/date.tsx [hunk 7]
```diff
@@ -232,14 +267,23 @@ export class Date {
...
 
+  deselectDate(day: Dayjs) {
+    const formattedDate = this.toDayString(day);
+    const currentValues = Array.from(this.value.split(','));
+
+    this.value = currentValues.filter(value => value !== formattedDate).join(',');
+    this.eventChange.emit(this.value);
+  }
+
```
Comment (dani-cl-madrid) [inline src/custom-elements/src/components/date/date.tsx:281]: If you create a method `addDate` this should be named `removeDate`. Makes sentido?

**Review by dani-cl-madrid — CHANGES_REQUESTED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/4/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://6fce31706b82637edc6bfe36fcfc703293993fda-chi-pr.enterprise.ctl.io

File: src/custom-elements/src/components/date/date.tsx (hunk 7 above)
Comment (janibolkvadze) [reply src/custom-elements/src/components/date/date.tsx:281]: Totally makes sentido. Will make the changes. Thanks Dani!

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/5/. :x:

**Review by mattnickles — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/9/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/11/. :x:

**Review by dani-cl-madrid — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/12/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/13/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/14/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://aa30daad3b10a52249b63a44a01f28049cfb2da2-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/15/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/16/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://938276553d1b9b830f34e732c05483c49566d777-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-24/17/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://938276553d1b9b830f34e732c05483c49566d777-chi-pr.enterprise.ctl.io

**Pull request merged**